src/main/java/uk/gov/justice/probation/courtcasematcher/service/SqsAdminService.java

```diff
@@ -0,0 +1,70 @@
+package uk.gov.justice.probation.courtcasematcher.service;
+
+import com.amazonaws.services.sqs.AmazonSQSAsync;
+import com.amazonaws.services.sqs.model.DeleteMessageRequest;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import static java.util.List.of;
+import static java.util.Optional.ofNullable;
+
+@Service
+@Slf4j
+public class SqsAdminService {
+    private static final String APPROXIMATE_NUMBER_OF_MESSAGES_ATTR_NAME = "ApproximateNumberOfMessages";
+
+    @Autowired
+    @Qualifier("courtCaseMatcherSqsQueue")
+    private AmazonSQSAsync courtCaseMatcherSqsDlq;
+
+    @Autowired
+    @Qualifier("courtCaseMatcherSqsDlq")
+    private AmazonSQSAsync courtCaseMatcherSqsQueue;
+
+
+    private String courtCaseMatcherSqsDlqUrl;
+
+    private String courtCaseMatcherSqsUrl;
+
+    public SqsAdminService(AmazonSQSAsync courtCaseMatcherSqsDlq, AmazonSQSAsync courtCaseMatcherSqsQueue,
+                           @Value("${aws_sqs_court_case_matcher_dlq_endpoint_url}") String courtCaseMatcherSqsDlqUrl,
+                           @Value("${aws_sqs_court_case_matcher_endpoint_url}") String courtCaseMatcherSqsUrl) {
+        this.courtCaseMatcherSqsDlq = courtCaseMatcherSqsDlq;
+        this.courtCaseMatcherSqsQueue = courtCaseMatcherSqsQueue;
+        this.courtCaseMatcherSqsDlqUrl = courtCaseMatcherSqsDlqUrl;
+        this.courtCaseMatcherSqsUrl = courtCaseMatcherSqsUrl;
+    }
+
+    public void replayDlqMessages() {
+
+        log.info("Replaying court case matcher dlq messages.");
+
+        var dlqAttributes = courtCaseMatcherSqsDlq.getQueueAttributes(courtCaseMatcherSqsDlqUrl,
+                of(APPROXIMATE_NUMBER_OF_MESSAGES_ATTR_NAME));
+        var messageCount = ofNullable(dlqAttributes.getAttributes())
+                .map(attrMap -> attrMap.get(APPROXIMATE_NUMBER_OF_MESSAGES_ATTR_NAME))
+                .map(Integer::parseInt)
+                .orElse(0);
+
+        log.info("Found {} messages on the dlq", messageCount);
+
+        var replayedMessages = 0;
+
+        while (replayedMessages < messageCount) {
```

Comment:
You could get rid of the while loop and use a `IntStream.range(0, messageCount).forEach()`  here 🤷  Personal preference really, I generally prefer streams but I'm fine with either here.